worldedit-core/src/main/java/com/fastasyncworldedit/core/FaweCache.java

```diff
@@ -137,6 +138,32 @@ public V load(@Nonnull T key) {
         });
     }
 
+    /**
+     * Create a new cache aimed to act as a thread-cache that is safe to the main server thread. If the method is called away
+     * from the main thread, it will return a {@link Function} referencing the {@link LoadingCache} returned by
+     * {@link FaweCache#createCache(Supplier)}. If it is called from the main thread, it will return a {@link Function} that
+     * will always return the result of the given {@link Supplier}. It is designed to prevent issues caused by
+     * interally-mutable and resettable classes such as {@link com.fastasyncworldedit.core.extent.filter.block.CharFilterBlock}
+     * from causing issues when used in edits on the main thread.
+     *
+     * @param withInitial The supplier used to determine the initial value if a thread cache is created, else to provide the
+     *                    a new instance of the class being cached if on the main thread.
+     * @return a {@link Function} referencing a cache, or the given {@link Supplier}
+     * @since TODO
+     */
+    @AnnotationHelper.ApiDescription(info = "Designed for specific internal use.")
```

Comment:
```suggestion
    @AnnotationHelper.ApiDescription(info = "Designed for specific internal use only.")
```

Reply:
I left out "only" has it could be a useful method for API use, however, it's advising that it's not really a general API method